packages/macro/src/macroJsx.ts

```diff
@@ -332,56 +334,69 @@ export default class MacroJSX {
+      if (_attr.isJSXSpreadAttribute()) {
         continue
       }
 
-      if (this.types.isJSXNamespacedName(attr.name)) {
+      const attr = _attr as NodePath<JSXAttribute>
+
+      if (this.types.isJSXNamespacedName(attr.node.name)) {
         continue
       }
 
-      const name = attr.name.name
+      const name = attr.node.name.name
+      const value = attr.get("value") as
+        | NodePath<Literal>
+        | NodePath<JSXExpressionContainer>
 
       if (name === "value") {
-        const exp = this.types.isLiteral(attr.value)
-          ? attr.value
-          : (attr.value as JSXExpressionContainer).expression
+        const exp = value.isLiteral() ? value : value.get("expression")
+
         token.name = this.expressionToArgument(exp)
-        token.value = exp as Expression
+        token.value = exp.node as Expression
       } else if (format !== "select" && name === "offset") {
         // offset is static parameter, so it must be either string or number
-        token.options.offset = this.types.isStringLiteral(attr.value)
-          ? attr.value.value
-          : ((attr.value as JSXExpressionContainer).expression as StringLiteral)
-              .value
+        token.options.offset =
+          value.isStringLiteral() || value.isNumericLiteral()
+            ? (value.node.value as string)
+            : (
+                (value as NodePath<JSXExpressionContainer>).get(
+                  "expression"
+                ) as NodePath<StringLiteral>
+              ).node.value
       } else {
-        let value: ArgToken["options"][number]
+        let option: ArgToken["options"][number]
 
-        if (this.types.isStringLiteral(attr.value)) {
-          value = (attr.value.extra.raw as string).replace(/(["'])(.*)\1/, "$2")
+        if (value.isStringLiteral()) {
+          option = (value.node.extra.raw as string).replace(
+            /(["'])(.*)\1/,
+            "$2"
+          )
         } else {
-          value = this.tokenizeChildren(attr.value)
+          option = this.tokenizeChildren(value as JSXChildPath)
         }
         if (pluralRuleRe.test(name)) {
-          token.options[jsx2icuExactChoice(name)] = value
+          token.options[jsx2icuExactChoice(name)] = option
         } else {
-          token.options[name] = value
+          token.options[name] = option
         }
       }
     }
 
     return token
   }
 
-  tokenizeElement = (node: JSXElement): ElementToken => {
+  tokenizeElement = (path: NodePath<JSXElement>): ElementToken => {
     // !!! Important: Calculate element index before traversing children.
     // That way outside elements are numbered before inner elements. (...and it looks pretty).
     const name = this.elementIndex()
-    const children = R.flatten(
-      node.children.map((child) => this.tokenizeChildren(child)).filter(Boolean)
-    )
+    const children = path
+      .get("children")
+      .flatMap((child) => this.tokenizeChildren(child))
+      .filter(Boolean)
```

Comment:
This logic is the same as in the `tokenizeTrans()` function. It would be good to have this logic only in one place.

Reply:
done